airbyte-webapp/src/components/JobItem/JobItem.tsx

```diff
@@ -32,28 +32,28 @@ const LoadLogs = styled.div`
 `;
 
 interface JobItemProps {
-  shortInfo?: boolean;
   job: SynchronousJobReadWithStatus | JobsWithJobs;
 }
 
 const didJobSucceed = (job: SynchronousJobReadWithStatus | JobsWithJobs) => {
   return getJobStatus(job) !== "failed";
 };
 
-export const getJobStatus: (job: SynchronousJobReadWithStatus | JobsWithJobs) => JobStatus = (job) => {
-  return (job as JobsWithJobs).job?.status ?? (job as SynchronousJobReadWithStatus).status;
+export const getJobStatus: (
+  job: SynchronousJobReadWithStatus | JobsWithJobs
+) => JobStatus | CheckConnectionReadStatus = (job) => {
+  return "status" in job ? job.status : job.job.status;
```

Comment:
:information_source: The casting also caused TypeScript to lose the real return type here, so changed this as well.